### PR TITLE
Remove old cronjob controller

### PR DESCRIPTION
**Pull request by soltysh**
#### What type of PR is this?
/kind cleanup
/sig apps

#### What this PR does / why we need it:
This is a followup to https://github.com/kubernetes/kubernetes/pull/104470, part of https://github.com/kubernetes/enhancements/issues/19

#### Special notes for your reviewer:
/assign @alaypatel07 @pacoxu 

#### Does this PR introduce a user-facing change?
```release-note
Remove deprecated and not supported old cronjob controller.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
- KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/19-Graduate-CronJob-to-Stable

### CODE REVIEW

File: pkg/controller/cronjob/utils.go [hunk 1]
```diff
@@ -232,28 +140,25 @@ func getMostRecentScheduleTime(earliestTime time.Time, now time.Time, schedule c
...
-// getJobFromTemplate makes a Job from a CronJob
-func getJobFromTemplate(cj *batchv1.CronJob, scheduledTime time.Time) (*batchv1.Job, error) {
-	labels := copyLabels(&cj.Spec.JobTemplate)
-	annotations := copyAnnotations(&cj.Spec.JobTemplate)
-	// We want job names for a given nominal start time to have a deterministic name to avoid the same job being created twice
-	name := fmt.Sprintf("%s-%d", cj.Name, getTimeHash(scheduledTime))
+// getTimeHash returns Unix Epoch Time
+func getTimeHash(scheduledTime time.Time) int64 {
+	return scheduledTime.Unix()
+}
```
Comment (pacoxu) [inline pkg/controller/cronjob/utils.go:146]: This method is not used anymore.

File: pkg/controller/cronjob/cronjob_controllerv2.go [hunk 2]
```diff
@@ -29,8 +29,10 @@ import (
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	ref "k8s.io/client-go/tools/reference"
```
Comment (pacoxu) [inline pkg/controller/cronjob/cronjob_controllerv2.go:35]: The CI failed for the import order is wrong here.

**Review by pacoxu — COMMENTED** [review]
Other parts are LGTM expect the ci failure and one unused method.

**Comment by soltysh** [comment]
@pacoxu all should be updated, ptal

**Pull request merged**